pkg/injection/namespace/mapper/mapper_test.go

```diff
@@ -11,28 +11,49 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func createTestDynakubeWithAppInject(name string, labels map[string]string, labelExpression []metav1.LabelSelectorRequirement) *dynatracev1beta2.DynaKube {
+func createBaseDynakube(name string, appInjection bool, metadataEnrichment bool) *dynatracev1beta2.DynaKube {
 	dk := &dynatracev1beta2.DynaKube{
 		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: "dynatrace"},
 		Spec: dynatracev1beta2.DynaKubeSpec{
-			OneAgent: dynatracev1beta2.OneAgentSpec{
-				ApplicationMonitoring: &dynatracev1beta2.ApplicationMonitoringSpec{},
+			MetadataEnrichment: dynatracev1beta2.MetadataEnrichment{
+				Enabled: metadataEnrichment,
 			},
 		},
 	}
-	if labels != nil {
-		dk.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector = metav1.LabelSelector{MatchLabels: labels}
-	}
 
-	if labelExpression != nil {
-		dk.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector = metav1.LabelSelector{MatchExpressions: labelExpression}
+	if appInjection {
+		dk.Spec.OneAgent = dynatracev1beta2.OneAgentSpec{
+			ApplicationMonitoring: &dynatracev1beta2.ApplicationMonitoringSpec{}}
 	}
 
 	return dk
 }
 
-func createTestDynakubeWithMultipleFeatures(name string, labels map[string]string) *dynatracev1beta2.DynaKube {
-	dk := createTestDynakubeWithAppInject(name, labels, nil)
+func createDynakubeWithAppInject(name string, selector metav1.LabelSelector) *dynatracev1beta2.DynaKube {
+	dk := createBaseDynakube(name, true, false)
+	dk.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector = selector
+
+	return dk
+}
+
+func createDynakubeWithMultipleFeatures(name string, selector metav1.LabelSelector) *dynatracev1beta2.DynaKube {
```

Comment:
do we need this anymore?

does the same as `createDynakubeWithAppInject`...?

Reply:
I agree, the name of it kinda sounds similar to what `createTestDynakubeWithMetadataAndAppInjection` does (that still has `Test` in the name 🙃 )

Reply:
good catch, find and replace is my kryptonite :D thanks!